### PR TITLE
Added vim-surround support for vim-mode and vim-mode-plus

### PR DESCRIPTION
**Pull request by geksilla**

### CODE REVIEW

File: src/cljs/proton/layers/core/core.cljs [hunk 1]
```diff
@@ -81,6 +81,14 @@
...
 
+(defmethod init-package [:core :vim-mode-plus] []
+  (atom-env/set-keymap! "atom-text-editor.vim-mode-plus.normal-mode"
+    {"y s" "vim-mode-plus:surround"
+     "d s" "vim-mode-plus:delete-surround"
+     "c s" "vim-mode-plus:change-surround"})
+  (atom-env/set-keymap! "atom-workspace atom-text-editor.vim-mode-plus.visual-mode"
+    {"s" "vim-mode-plus:surround"}))
+
 (defmethod get-keybindings :core [] @keybindings)
```
Comment (robbyoconnor) [inline src/cljs/proton/layers/core/core.cljs]: How does this work for vim-mode?

**Comment by dvcrn** [comment]
Do you think we could leverage the `get-keymaps` multi for this?

**Comment by geksilla** [comment]
@dvcrn, yes we can, but using `init-package` hook appropriate key maps will be added only when `vim-mode-plus` package enabled.

**Comment by dvcrn** [comment]
Hmm, we could check which provider is set and then push into the `keymaps` atom.

I'm just worried that if we start this way, we will start editing things more direct and skip the collect phase. Right now we don't need it, but in the future we could for example pull the keymap out of the multi and stuff it in a README or something like that

Nevertheless, LGTM!

**Pull request merged**